linode_api4/groups/object_storage.py

```diff
@@ -0,0 +1,616 @@
+            raise ValueError(
+                "Content-type header is missing for the current method! It's required for all methods except GET or DELETE."
+            )
+        params = {
+            "method": method,
+            "name": name,
+            "expires_in": expires_in,
+            "content_type": content_type,
+        }
+
+        result = self.client.post(
+            "/object-storage/buckets/{}/{}/object-url".format(
+                cluster_id, bucket
+            ),
+            data=drop_null_keys(params),
+        )
+
+        if not "url" in result:
+            raise UnexpectedResponseError(
+                "Unexpected response when creating the access url of an object!",
+                json=result,
+            )
+
+        return MappedObject(**result)
+
+    def ssl_cert_delete(self, cluster_id, bucket):
+        """
+        Deletes this Object Storage bucket’s user uploaded TLS/SSL certificate
+        and private key.
+
+        API Documentation: https://www.linode.com/docs/api/object-storage/#object-storage-tlsssl-cert-delete
+
+        :param cluster_id: The ID of the cluster this bucket exists in.
+        :type cluster_id: str
+
+        :param bucket: The bucket name.
+        :type bucket: str
+
+        :returns: True if the TLS/SSL certificate and private key in the bucket were successfully deleted.
+        :rtype: bool
+        """
+
+        resp = self.client.delete(
+            "/object-storage/buckets/{}/{}/ssl".format(cluster_id, bucket)
+        )
+
+        if "error" in resp:
+            raise UnexpectedResponseError(
+                "Unexpected response when deleting a bucket!",
+                json=resp,
+            )
+        return True
+
+    def ssl_cert(self, cluster_id, bucket):
+        """
+        Returns a result object which wraps a bool value indicating
+        if this bucket has a corresponding TLS/SSL certificate that
+        was uploaded by an Account user.
+
+        API Documentation: https://www.linode.com/docs/api/object-storage/#object-storage-tlsssl-cert-view
+
+        :param cluster_id: The ID of the cluster this bucket exists in.
+        :type cluster_id: str
+
+        :param bucket: The bucket name.
+        :type bucket: str
+
+        :returns: A result object which has a bool field indicating if this Bucket has a corresponding
+                  TLS/SSL certificate that was uploaded by an Account user.
+        :rtype: MappedObject
+        """
+        result = self.client.get(
+            "/object-storage/buckets/{}/{}/ssl".format(cluster_id, bucket)
+        )
+
+        if not "ssl" in result:
+            raise UnexpectedResponseError(
+                "Unexpected response when getting the TLS/SSL certs indicator of a bucket!",
+                json=result,
+            )
+
+        return MappedObject(**result)
+
+    def ssl_cert_upload(self, cluster_id, bucket, certificate, private_key):
```

Comment:
For consistency, we might want to move these under the `ObjectStorageBucket` object since they manage TLS/SSL certs for an individual bucket 👍